### PR TITLE
Add DOTImporter

### LINKED ISSUE #127: DOT Import
We need a DOT file import method to go with the export functionality that currently exists.

I have written a version which works for our cases and will be submitting a pull request shortly.

### PR DESCRIPTION
**Pull request by wselwood**
Import class to read DOT files. This should resolve #127 

One thing I am not sure about is if I have put it in the right place. I couldn't find any existing import functions to put it with so I created it along side the export classes.

### CODE REVIEW

File: jgrapht-ext/src/main/java/org/jgrapht/ext/DOTImporter.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+   //  type="SeenIn" last-seen="2015-04-07T14:30:00.000-0100"
+   private Map<String, String> splitAttributes(String input)
+   {
+      int index = 0;
+      Map<String, String> result = new HashMap<String, String>();
+      while(index < input.length()) {
+         int nextEquals = input.indexOf('=', index);
+         String key = input.substring(index, nextEquals).trim();
+         int firstQuote = input.indexOf('\"', nextEquals) + 1;
+         int secondQuote = input.indexOf('\"', firstQuote);
```
Comment (jsichi) [inline jgrapht-ext/src/main/java/org/jgrapht/ext/DOTImporter.java]: This needs to deal with backslash-escaped quotation marks; see

http://www.graphviz.org/doc/info/lang.html

File: jgrapht-ext/src/main/java/org/jgrapht/ext/DOTImporter.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+               label = id;
+            }
+
+            V existing = vertexes.get(id);
+            if (existing == null) {
+               V vertex = vertexProvider.buildVertex(label, attributes);
+               graph.addVertex(vertex);
+               vertexes.put(id, vertex);
+            } else {
+              throw new ImportException("out of order input");
```
Comment (jsichi) [inline jgrapht-ext/src/main/java/org/jgrapht/ext/DOTImporter.java]: Is this restriction part of the DOT language?

File: jgrapht-ext/src/main/java/org/jgrapht/ext/DOTImporter.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+         throw new ImportException("Dot string was invalid");
+      }
+      // validate the first line
+      String[] firstLine = lines[0].split(" ", 3);
+      if(firstLine.length != 3) {
+         throw new ImportException("not enough parts on first line");
+      }
+
+      if (!firstLine[0].equals("digraph") && !firstLine[0].equals("graph")) {
+         throw new ImportException("unknown graph type");
```
Comment (jsichi) [inline jgrapht-ext/src/main/java/org/jgrapht/ext/DOTImporter.java]: We could also validate to make sure that directed/undirected declaration matches the graph being built?

**Comment by jsichi** [comment]
Thanks, the location is fine.  All the files need the standard JGraphT headers, and all classes, interfaces, and public methods need Javadoc.

**Comment by wselwood** [comment]
Hi @jsichi 

I've made changes based on your comments and a few other bits I spotted. Thanks for the feedback. Let me know if you have any more comments.

Not sure what happened with the build. jdk 7 and 8 have passed.  But the openjdk-6 build seems to have got a version of maven that doesn't support it. Running on my machine with a 1.6.0_34 open jdk passes. Do you have any ideas?

**Comment by jsichi** [comment]
Looks good, I've merged it.

My bad on the build; we've dropped support for JDK 6, but I forgot to update the Travis configuration.  That's done now.

**Pull request merged**